catroid/src/main/java/org/catrobat/catroid/web/requests/HttpRequests.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2019 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.web.requests
+
+import android.util.Log
+import com.squareup.okhttp.MediaType
+import com.squareup.okhttp.MultipartBuilder
+import com.squareup.okhttp.Request
+import com.squareup.okhttp.RequestBody
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.common.FlavoredConstants
+import org.catrobat.catroid.transfers.project.ProjectUploadData
+import org.catrobat.catroid.transfers.project.UPLOAD_FILE_NAME
+import org.catrobat.catroid.utils.Utils
+import org.catrobat.catroid.web.ServerCalls
+import org.catrobat.catroid.web.ServerCalls.BASE_URL_TEST_HTTPS
+
+private const val FILE_UPLOAD_TAG = "upload"
+private const val PROJECT_NAME_TAG = "projectTitle"
+private const val PROJECT_DESCRIPTION_TAG = "projectDescription"
+private const val PROJECT_CHECKSUM_TAG = "fileChecksum"
+private const val USER_EMAIL = "userEmail"
+private const val DEVICE_LANGUAGE = "deviceLanguage"
+private val MEDIA_TYPE_ZIPFILE = MediaType.parse("application/zip")
+private const val FILE_UPLOAD_URL = FlavoredConstants.BASE_URL_HTTPS + "api/upload/upload.json"
+private const val TEST_FILE_UPLOAD_URL_HTTP = BASE_URL_TEST_HTTPS + "api/upload/upload.json"
+
+private val serverUrl: String
+    get() {
+        return if (ServerCalls.useTestUrl) {
+            TEST_FILE_UPLOAD_URL_HTTP
```

Comment:
hmm, I know it's always difficult with testing, but could there be an other (quick) way to inject the testing URL? the production code should not know about test stuff.

Reply:
this is basically just refactored from the java source code... After having a quick look at where the useTestUrl flag is in the project, I'd rather not fix this within this ticket, but write a separate refactoring story for this.